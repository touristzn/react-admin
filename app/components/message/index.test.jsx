import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Message from './index'

/**
 * 引用path：@/static/utils/message.js
 */

let wrapper

describe('<Message />', () => {
  const props = {
    // options可以传对象，也可以直接传字符串
    options: {
      // success、info、warning、danger
      type: 'success', 
      message: '提示内容'
    }
  }

  beforeEach(() => {
    wrapper = shallow(
      <Message {...props} />
    )
  })

  it('renders the text and contain the class', () => {
    expect(wrapper.text()).toBe(props.options.message)
    expect(wrapper.hasClass('success')).toEqual(true)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})