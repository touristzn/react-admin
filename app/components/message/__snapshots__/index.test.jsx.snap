// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders correctly 1`] = `
<div
  className="message slideInDown success"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  提示内容
</div>
`;
